### ISSUE TITLE
uppercase extension failing

### ISSUE THREAD

**Issue by roelpaesen** [loris-imageserver/loris#258]
Hi,
when I have a file 'file.jpg', and I use the url "<server>/file.jpg/full/full/0/default.jpg" everything works great.

Now, when I copy 'file.jpg' to 'file.JPG' (ext4 filesystem, so these are seperate and valid files) and change the url accordingly to
"<server>/file.JPG/full/full/0/default.jpg"
a 500-internal server error occurs.
(when then remove FILE.JPG and browse to "<server>/file.JPG/full/full/0/default.jpg" I get an error that the resource is not found:
"Not Found: Source image not found for identifier: <identifier>"

So, is this an error in my configuration somewhere, or is there a bug in loris?
Thanks already for looking into this!

version used: git development branch as of 2016-11-22

And because logging is sometimes usefull :)

```
[Tue Nov 22 17:01:30.530010 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891] mod_wsgi (pid=1340): Exception occurred processing WSGI script '/var/www/loris2/loris2.wsgi'.
[Tue Nov 22 17:01:30.530235 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891] Traceback (most recent call last):
[Tue Nov 22 17:01:30.530300 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]   File "/usr/local/lib/python2.7/dist-packages/Loris-2.0.0-py2.7.egg/loris/webapp.py", line 384, in __call__
[Tue Nov 22 17:01:30.530781 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]     return self.wsgi_app(environ, start_response)
[Tue Nov 22 17:01:30.530807 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]   File "/usr/local/lib/python2.7/dist-packages/Loris-2.0.0-py2.7.egg/loris/webapp.py", line 326, in wsgi_app
[Tue Nov 22 17:01:30.530844 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]     response = self.route(request)
[Tue Nov 22 17:01:30.530858 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]   File "/usr/local/lib/python2.7/dist-packages/Loris-2.0.0-py2.7.egg/loris/webapp.py", line 369, in route
[Tue Nov 22 17:01:30.530877 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]     return self.get_img(request, ident, region, size, rotation, quality, fmt, base_uri)
[Tue Nov 22 17:01:30.530889 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]   File "/usr/local/lib/python2.7/dist-packages/Loris-2.0.0-py2.7.egg/loris/webapp.py", line 548, in get_img
[Tue Nov 22 17:01:30.530905 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]     info = self._get_info(ident, request, base_uri, src_fp, src_format)[0]
[Tue Nov 22 17:01:30.530916 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]   File "/usr/local/lib/python2.7/dist-packages/Loris-2.0.0-py2.7.egg/loris/webapp.py", line 464, in _get_info
[Tue Nov 22 17:01:30.531010 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]     formats = self.transformers[src_format].target_formats
[Tue Nov 22 17:01:30.531079 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891] KeyError: 'JPG'
```

**Comment by roelpaesen** [comment]
And because logging is sometimes usefull :)

```
[Tue Nov 22 17:01:30.530010 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891] mod_wsgi (pid=1340): Exception occurred processing WSGI script '/var/www/loris2/loris2.wsgi'.
[Tue Nov 22 17:01:30.530235 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891] Traceback (most recent call last):
[Tue Nov 22 17:01:30.530300 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]   File "/usr/local/lib/python2.7/dist-packages/Loris-2.0.0-py2.7.egg/loris/webapp.py", line 384, in __call__
[Tue Nov 22 17:01:30.530781 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]     return self.wsgi_app(environ, start_response)
[Tue Nov 22 17:01:30.530807 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]   File "/usr/local/lib/python2.7/dist-packages/Loris-2.0.0-py2.7.egg/loris/webapp.py", line 326, in wsgi_app
[Tue Nov 22 17:01:30.530844 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]     response = self.route(request)
[Tue Nov 22 17:01:30.530858 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]   File "/usr/local/lib/python2.7/dist-packages/Loris-2.0.0-py2.7.egg/loris/webapp.py", line 369, in route
[Tue Nov 22 17:01:30.530877 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]     return self.get_img(request, ident, region, size, rotation, quality, fmt, base_uri)
[Tue Nov 22 17:01:30.530889 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]   File "/usr/local/lib/python2.7/dist-packages/Loris-2.0.0-py2.7.egg/loris/webapp.py", line 548, in get_img
[Tue Nov 22 17:01:30.530905 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]     info = self._get_info(ident, request, base_uri, src_fp, src_format)[0]
[Tue Nov 22 17:01:30.530916 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]   File "/usr/local/lib/python2.7/dist-packages/Loris-2.0.0-py2.7.egg/loris/webapp.py", line 464, in _get_info
[Tue Nov 22 17:01:30.531010 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891]     formats = self.transformers[src_format].target_formats
[Tue Nov 22 17:01:30.531079 2016] [wsgi:error] [pid 1340:tid 139873742907136] [remote 172.16.10.10:6891] KeyError: 'JPG'
```

**Comment by jpstroop** [comment]
I think you hit on it--I'm sorry I didn't get to respond earlier. The application uses the file extension to know which Transformer instance to use to derive the new file. See here:

https://github.com/loris-imageserver/loris/blob/development/loris/webapp.py#L636

and the default resolver (cheaply) just takes the extension and trusts that it's the format:

https://github.com/loris-imageserver/loris/blob/development/loris/resolver.py#L90-L91

I don't think there would be any harm in having the resolver normalize that to lower case.

**Comment by roelpaesen** [comment]
I tested changing line 91 of resolver.py from
```
return ident.split('.')[-1]
```
to
```
return ident.split('.')[-1].lower()
```
and now it works :)

**Comment by bcail** [comment]
This should be fixed with PR #260. Closing.
